The first header-check resolves AC_REQUIRE(AC_PROG_EGREP). If that is not
run due to some condition, EGREP is unset for subsequent checks.
This breaks the checks for function prototypes.

--- mico/configure.in
+++ mico/configure.in
@@ -411,6 +411,8 @@
 AC_SUBST(EXTRA_CXXFLAGS)
 ac_compile="$ac_compile "'$EXTRA_CXXFLAGS'
 
+AC_HEADER_STDC
+
 #
 # thread support
 #
@@ -1342,7 +1344,6 @@
 # Checks for header files.
 #
 
-AC_HEADER_STDC
 AC_CHECK_HEADERS(fcntl.h unistd.h sys/select.h strings.h float.h ieeefp.h)
 AC_CHECK_HEADERS(sys/un.h netinet/in.h arpa/inet.h netdb.h dlfcn.h dl.h)
 AC_CHECK_HEADERS(netinet/tcp.h stdlib.h sys/time.h sunmath.h sys/stat.h)
